frontend/src/page-context.ts

```diff
@@ -1,7 +1,23 @@
 interface tokenResponse {
-   socketToken: string
+   socketToken: string,
+   user: string,
 }
-export default function getPageContext(): Promise<tokenResponse> {
+
+let pageContext: tokenResponse;
+
+if (process.env.DUMMY_USER) {
+   pageContext = {
+      user: process.env.DUMMY_USER,
+      socketToken: "fake-token",
+   };
+}
+
+export function getUser() {
+   return pageContext && pageContext.user;
+}
+
+export function getPageContext(): Promise<tokenResponse> {
```

Comment:
Are you just saying this because it has *context* in the name? :trollface: 

Seriously though, this is used in three places and only one of which is a component. That said, I understand that it changing values wouldn't cause the components to re-render. In this case, we have to retrieve the page context *before* we receive the list of pulls, so the values never change over the life of the page.

Though if we add more functionality, expose more data, or allow that to change, a context will definitely be the right choice

Reply:
It's totally reasonable to have it like it is. I happen to like contexts because they are stack-sensitive, so each one only affects the components within it, and can be overridden by other instances lower down the stack. Also, I feel good about them as a global variable alternative because I see the Racket folks using them, and they usually seem to have clever ideas.